From: Eneas U de Queiroz <cotequeiroz@gmail.com>
Date: Tue, 18 Feb 2020 17:47:20 -0300
Subject: kernel: avoid underscore in *6lowpan package names

Packages kmod-bluetooth_6lowpan and kmod-ieee802154_6lowpan contain an
underscore in the package name.  This causes problems in package/install
because when building a list of package files to install offline using
opkg, it uses a wildcard of the form $(dir)/$(pkg)_*.ipk.

If you were to select kmod-bluetooth=y, but kmod-bluetooth_6lowpan=m,
the latter would be picked up by that wildcard, and make package/install
would fail:

Collected errors:
 * satisfy_dependencies_for: Cannot satisfy the following dependencies
 * for kmod-bluetooth_6lowpan:
 *      kmod-6lowpan
 * opkg_install_cmd: Cannot install package kmod-bluetooth_6lowpan.

Changing the wildcard pattern is not trivial, and there may be other
places in the build system making this assumption about the package name
format.

Using a dash in place of the underscore avoids the issue.

Signed-off-by: Eneas U de Queiroz <cotequeiroz@gmail.com>

diff --git a/package/kernel/linux/modules/other.mk b/package/kernel/linux/modules/other.mk
index 16b438461e58d4be1f190e3b49e363e690110515..0e3c98e40d5a1b936539a5c09bc6cd2f0d39fd93 100644
--- a/package/kernel/linux/modules/other.mk
+++ b/package/kernel/linux/modules/other.mk
@@ -87,7 +87,7 @@ endef
 $(eval $(call KernelPackage,ath3k))
 
 
-define KernelPackage/bluetooth_6lowpan
+define KernelPackage/bluetooth-6lowpan
   SUBMENU:=$(OTHER_MENU)
   TITLE:=Bluetooth 6LoWPAN support
   DEPENDS:=+kmod-6lowpan +kmod-bluetooth
@@ -96,11 +96,11 @@ define KernelPackage/bluetooth_6lowpan
   AUTOLOAD:=$(call AutoProbe,bluetooth_6lowpan)
 endef
 
-define KernelPackage/bluetooth_6lowpan/description
+define KernelPackage/bluetooth-6lowpan/description
  Kernel support for 6LoWPAN over Bluetooth Low Energy devices
 endef
 
-$(eval $(call KernelPackage,bluetooth_6lowpan))
+$(eval $(call KernelPackage,bluetooth-6lowpan))
 
 
 define KernelPackage/btmrvl
diff --git a/package/kernel/linux/modules/wpan.mk b/package/kernel/linux/modules/wpan.mk
index 1e986ee0ca8ebe663175ada97a839945736762c6..0c2b0136b79ccc661cb9af8e867ba2059b19fc95 100644
--- a/package/kernel/linux/modules/wpan.mk
+++ b/package/kernel/linux/modules/wpan.mk
@@ -116,7 +116,7 @@ endef
 
 $(eval $(call KernelPackage,cc2520))
 
-define KernelPackage/ieee802154_6lowpan
+define KernelPackage/ieee802154-6lowpan
   SUBMENU:=$(WPAN_MENU)
   TITLE:= 6LoWPAN support over IEEE-802.15.4
   DEPENDS:=+kmod-6lowpan +kmod-ieee802154
@@ -127,8 +127,8 @@ define KernelPackage/ieee802154_6lowpan
   AUTOLOAD:=$(call AutoLoad,91,ieee802154_6lowpan)
 endef
 
-define KernelPackage/ieee802154_6lowpan/description
+define KernelPackage/ieee802154-6lowpan/description
  IPv6 compression over IEEE 802.15.4
 endef
 
-$(eval $(call KernelPackage,ieee802154_6lowpan))
+$(eval $(call KernelPackage,ieee802154-6lowpan))
